src/test/mod.rs

```diff
@@ -123,7 +121,13 @@ impl TestEnvironment {
     }
 
     fn base_config(&self) -> Config {
-        Config::from_env().expect("failed to get base config")
+        let mut config = Config::from_env().expect("failed to get base config");
+
+        // Use less connections for each test compared to production.
+        config.max_pool_size = 2;
```

Comment:
2 seems a little small, don't we want as many connections as we have cores so that tests can run in parallel?
```suggestion
        // Use less connections for each test compared to production.
        config.max_pool_size = 32;
```

Reply:
That's the connection pool for each test, not the connection pool for the whole `cargo test`.